### DISCUSSION TITLE
"data-testid" in ConfirmModal confirmProps?

### DISCUSSION THREAD

**Discussion by ftzi** [mantinedev/mantine#3179] [Feature requests]
I can't use `data-testid` in openConfirmModal button props.

```tsx
modals.openConfirmModal({
  title: 'Delete?',
  labels: { cancel: 'Cancel', confirm: 'Delete' },
  confirmProps: {
    color: 'red',
    'data-testid': 'confirm-delete' // TS throws the error below
  },
})
```

> Type '{ color: string; 'data-testid': string; }' is not assignable to type 'ButtonProps & Pick<DetailedHTMLProps<ButtonHTMLAttributes<HTMLButtonElement>, HTMLButtonElement>, "key" | keyof ButtonHTMLAttributes<...>>'.
>   Object literal may only specify known properties, and ''data-testid'' does not exist in type 'ButtonProps & Pick<DetailedHTMLProps<ButtonHTMLAttributes<HTMLButtonElement>, HTMLButtonElement>, "key" | keyof ButtonHTMLAttributes<...>>'.ts(2322)

What I've been doing is this:
```tsx
confirmProps: {
  color: 'red',
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  ...({ 'data-testid': 'confirm-delete' } as any),
},
```

but it's quite hacky and lengthy. Note that I have an intentional ESLint rule that forbids explicit any type.

Is there any built-in way to do it? I can make a PR for it if necessary.

**Comment by rtivital** [comment]
I cannot think of any way of resolving this issue

**Reply by ftzi** [reply]
I just did

```tsx
declare module '@mantine/core' {
  export interface ButtonProps {
    'data-testid'?: string
  }
}
```

in a file and it works. Maybe it's the best thing to do as my case doesn't seem to be common.

**Reply by grahampcharles** [reply]
Your case might not be common but I use `data-autoFocus` on `confirmProps` almost universally, so I appreciate this insight.
